uma/protocol.py

```diff
@@ -459,8 +495,36 @@ def to_dict(self) -> Dict[str, Any]:
     @classmethod
     def _from_dict(cls, json_dict: Dict[str, Any]) -> Dict[str, Any]:
         return {
-            "signing_pubkey": bytes.fromhex(json_dict["signingPubKey"]),
-            "encryption_pubkey": bytes.fromhex(json_dict["encryptionPubKey"]),
+            "signing_cert_chain": (
+                [
+                    x509.load_pem_x509_certificate(bytes.fromhex(cert))
+                    for cert in json_dict["signingCertChain"]
+                ]
+                if "signingCertChain" in json_dict
+                and json_dict["signingCertChain"] is not None
```

Comment:
Is this line needed if the previous is true?

Reply:
ooh good point! updated